src/commands/status/masternode.js

```diff
@@ -1,40 +1,84 @@
 const { table } = require('table');
+const fetch = require('node-fetch');
 
 const BaseCommand = require('../../oclif/command/BaseCommand');
+const CoreService = require('../../core/CoreService');
 
 class MasternodeStatusCommand extends BaseCommand {
   /**
    * @param {Object} args
    * @param {Object} flags
    * @param {DockerCompose} dockerCompose
+   * @param {CoreService} coreService
    * @param {Config} config
    * @return {Promise<void>}
    */
   async runWithDependencies(
     args,
     flags,
     dockerCompose,
+    createRpcClient,
     config,
   ) {
     const rows = [];
 
+    const coreService = new CoreService(
+      createRpcClient(),
+      dockerCompose.docker.getContainer('core')
+    );
+
     // Version
-    const versionOutput = await dockerCompose.execCommand(
+    const dashdVersion = (await dockerCompose.execCommand(
       config.toEnvs(),
       'core',
       'dashd --version',
-    );
+    )).out.split('\n')[0];
 
-    rows.push(['Version', versionOutput.out.split('\n')[0]]);
+    const mnsyncStatus = (await coreService.getRpcClient().mnsync('status')).result;
+    const blockchainInfo = (await coreService.getRpcClient().getBlockchainInfo()).result;
+    const masternodeStatus = (await coreService.getRpcClient().masternode('status')).result;
 
-    // Block count
-    const blockCountOutput = await dockerCompose.execCommand(
+    // Sentinel
+    const sentinelState = (await dockerCompose.execCommand(
       config.toEnvs(),
-      'core',
-      'dash-cli getblockcount',
-    );
+      'sentinel',
+      'python bin/sentinel.py',
+    )).out.split('\n')[0];
+
+    // Tendermint
+    const tendermintVersion = (await dockerCompose.execCommand(
+      config.toEnvs(),
+      'drive_tendermint',
+      'tendermint version',
+    )).out.split('\n')[0];
+
+    const tendermintStatus = JSON.parse((await dockerCompose.execCommand(
+      config.toEnvs(),
+      'drive_tendermint',
+      'curl localhost:26657/status',
+    )).out);
+
+    // Port check
+    const portState = await fetch('https://mnowatch.org/' + config.options.core.p2p.port + '/').then(res => res.text());
 
-    rows.push(['Blocks', blockCountOutput.out.trim()]);
+    // Build table
+    rows.push(['Dashd Version', dashdVersion]);
+    rows.push(['Sync Status', mnsyncStatus.AssetName]);
+    rows.push(['Headers', blockchainInfo.headers]);
+    rows.push(['Blocks', blockchainInfo.blocks]);
+    rows.push(['Masternode Status', masternodeStatus.state]);
+    if (masternodeStatus.state === 'READY') {
+      rows.push(['ProTx Hash', masternodeStatus.proTxHash]);
+      rows.push(['Service', masternodeStatus.dmnState.service]);
+      rows.push(['PoSe Penalty', masternodeStatus.dmnState.PoSePenalty]);
+    }
+    rows.push(['Sentinel', (sentinelState !== '' ? sentinelState : 'No errors')]);
+    rows.push(['Port Check', config.options.core.p2p.port + ' ' + portState]);
```

Comment:
Can we do port check for Tendermint too?

Reply:
Yes, but I think we need to change the config schema so it contains the host port for Tendermint P2P first, otherwise the port will be hardcoded here like I have done now. This is also preventing us from running multiple evonet nodes on one host at the moment. Should I try a PR for this?

Reply:
Yep, we can do that ;)

Reply:
I have implemented it to read the tendermint port from docker inspect instead, so we can support both current and future tendermint config variants.